### PR TITLE
bsunanda:Run2-hcx31 Add simulation code for HGCal and single layer fast timing device

### PR DESCRIPTION
**Pull request by bsunanda**
Code of detectors only for Phase 2.

### CODE REVIEW

**Comment by lgray** [comment]
@cmsbuild please test

@sethzenz @kpedro88 @vandreev11 @pfs

**Comment by civanch** [comment]
@bsunanda , why you have integer

HGCSD = cms.PSet(
...
+        TimeSliceUnit    = cms.int32(1),
....

but double time slices for old detectors?

**Comment by lgray** [comment]
@bsunanda Please do ```runTheMatrix.py -s -i all -j 8``` to check that things run before you submit pull requests. Here, though, it looks  like a commit was forgotten, perhaps?

**Comment by bsunanda** [comment]
I did the runTheMatrix - I do not understand how it misses these. Now the corrected version is committed
________________________________

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by lgray** [comment]
@civanch @mdhildreth Would it be possible to prioritize the review of this PR? It is important for the HGCal development timeline. Thanks!

**Comment by civanch** [comment]
+1

File: SimG4CMS/Calo/interface/CaloHitID.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ class CaloHitID {
   int          theTrackID;
   int          theTimeSliceID;
   uint16_t     theDepth;
-  int          timeSliceUnit;
+  double       timeSliceUnit;
```
Comment (davidlange6) [inline SimG4CMS/Calo/interface/CaloHitID.h]: @bsunanda - could this be a float?

Comment (bsunanda) [inline SimG4CMS/Calo/interface/CaloHitID.h]: It is better be a double (in any case this is a transient object)
________________________________

**Comment by davidlange6** [comment]
@cmsbuild, please test

**Comment by lgray** [comment]
@civanch, @mdhildreth Please recheck and resign this PR as you are able. Thanks!

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**